src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpMethodRepoDAO.scala

```diff
@@ -0,0 +1,43 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import akka.actor.ActorSystem
+import org.broadinstitute.dsde.rawls.model.AgoraEntity
+import org.broadinstitute.dsde.rawls.model.WorkspaceJsonSupport._
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.util.{Success,Failure,Try}
+import spray.client.UnsuccessfulResponseException
+import spray.client.pipelining._
+import spray.http.{StatusCodes, HttpCookie}
+import spray.http.HttpHeaders.Cookie
+import spray.httpx.SprayJsonSupport._
+
+/**
+ * @author tsharpe
+ */
+class HttpMethodRepoDAO( methodRepoServiceURL: String )( implicit system: ActorSystem ) extends MethodRepoDAO {
+
+  override def getMethodConfig( namespace: String, name: String, version: String, authCookie: HttpCookie ): Option[AgoraEntity] = {
+    val url = s"${methodRepoServiceURL}/configurations/${namespace}/${name}/${version}"
+    import system.dispatcher
+    val pipeline = addHeader(Cookie(authCookie)) ~> sendReceive ~> unmarshal[AgoraEntity]
+    Try(Await.result(pipeline(Get(url)),Duration.Inf)) match {
+      case Success(entity) => Option(entity)
+      case Failure(notOK: UnsuccessfulResponseException) =>
+        if ( notOK.responseStatus == StatusCodes.NotFound ) None
+        else throw notOK
+      case Failure(exception) => throw exception }
+  }
+
+  override def getMethod( namespace: String, name: String, version: String, authCookie: HttpCookie ): Option[AgoraEntity] = {
```

Comment:
Since the internal logic of ```getMethodConfig``` and ```getMethod``` isz so similar, could we factor some of it out into a private helper method?